[TC-346] MKR WiFi 1010 App Note (#485)
arduino/docs-content: content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md

---

> title: 'Understanding Battery Capabilities Of The MKR 1010 WiFi'

Reviewer: I think this summarizes the article better.

```suggestion
title: 'MKR WiFi 1010 Battery Application Note'

```

difficulty: intermediate

> description: 'Choose the correct battery for you MKR application.'

Reviewer: ```suggestion
description: 'Learn about different topics related to batteries, and how to choose the correct battery for you MKR application.'

```

tags:
  - Battery
  - MKR
  - 
author: 'Ali Jahangiri'
hardware:
  - hardware/01.mkr/01.boards/mkr-wifi-1010
software:
  - Arduino IDE 2.x
---

## Introduction

While you can run your MKR board from USB, you can make best use of it when your MKR board is able to operate independently of your computer and a USB cable. This is particularly useful for many IoT applications, use of relays with high current draw and remote sensing. Batteries can help us in this regard and a special connector is included in the MKR products for this end. In this application note, we will take a closer look at the battery capabilities of the MKR boards. 

### Goals
The goals of this project are to:

> - Understand what the terms `charging capacity` and `charging current`

Reviewer: Use backticks only for code. When highlighting something like this, let's bold them.

```suggestion
- Understand what the terms **charging capacity** and **charging current** mean.

```

Author: Noted!

- Distinguish between LiPo and Li-Ion batteries
- Learn how to identify connector polarity
- Understand how the SAMD chip reads battery voltage via a voltage divider

### Hardware & Software Needed
- [Arduino MKR WiFi 1010](https://store.arduino.cc/products/arduino-mkr-wifi-1010)
- Multimeter

> - IDE 2.x

Reviewer: ```suggestion
- [Arduino IDE](https://www.arduino.cc/en/software)

```

The BQ24195L IC is responsible for charging the single cell lion battery, as well as boosting the battery voltage to 3.3V for use on the board internally. An Arduino library ([Arduino_BQ24165](https://github.com/arduino-libraries/Arduino_BQ24195)) has been developed, to help control the features of this battery over I2C. 

### Electrical Properties

**Voltage**
The nominal voltage of both LiPo and Li-Ion batteries is around 3.7V and is commonly referred to as the voltage of single cell batteries. However, this is an ideal scenario since as a battery is used the voltage changes. The voltage can be considered similar to a waterfall. The higher the waterfall, the higher the voltage.

In the MKR boards, the battery terminal is connected to the SAMD21 via a reserved pin (PB09) known as `ADC_BATTERY` within the Arduino Core. This pin is used internally on the board, and is not accessible via the MKR pins. Since the voltage of a Li-ion battery exceeds 3.3V (the AREF value), a voltage divider must be used to extend the range while also ensure that only safe voltages are applied to the microcontroller. We can calculate the output voltage using the following formula

> $$ V_{out} = \frac{V_{source} \times R2} {R_1 + R_2} $$

Reviewer: Hey @aliphys. Nicely written document for the MKR board. But I think I discovered a little bug here, which is the use of "$" as a styling component? It doesn't render anything at docs if you take a look at it, so I am not sure how you can render it differently? Like using `backticks` maybe?

Author: Thanks for catching the render issue. The `$` sign is used to render math in Tex. It renders OK locally.

@jhansson-ard What syntax do we use for Tex in the `docs-content` repo?

```arduino
voltBat = voltADC * (max_Source_voltage/3.3);
```

**15.** We can approximate the battery voltage to be proportional to the capacity level. Since the `map()` function does not work with float variables, we will manually map the values.
```arduino
int new_batt = (voltBat - batteryEmptyVoltage) * (100) / (batteryFullVoltage - batteryEmptyVoltage);
```

**16.** We can now send the obtained values over Serial.
```arduino
  Serial.print("The ADC on PB09 reads a value of ");
  Serial.print(rawADC);
  Serial.print(" which is equivialent to ");
  Serial.print(voltADC);
  Serial.print("V. This means the battery voltage is ");
  Serial.print(voltBat);
  Serial.print("V. Which is equivalent to a charge level of ");
  Serial.print(new_batt);
  Serial.println("%.");
```

**17.** Finally, we will add a half a second delay at the end of the `loop()` function to allow variables to come slowly through the Serial Monitor.
```arduino
  delay(500);
}
```

**18.** You section for the `loop()` function should look as follows.
![Battery level calculation and Serial printing](assets/calculation-serial-printing.png)

> **11.** The complete code (with the addition of comments) is as follows. You can copy and paste this directly into your IDE

Author: ```suggestion
**19.** The complete code (with the addition of comments) is as follows. You can copy and paste this directly into your IDE

```
